### PR TITLE
Support favicon customization

### PR DESCRIPTION
**Pull request by omarmoo5**
Solves #519 .

### CODE REVIEW

File: starlette_admin/templates/base.html [hunk 1]
```diff
@@ -4,6 +4,9 @@
 <head>
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1, viewport-fit=cover"/>
+    {% if favicon_url %}
+        <link rel="icon" href="{{ favicon_url }}"/>
```
Comment (disrupted) [inline starlette_admin/templates/base.html:8]: consider setting the type as well
```suggestion
        <link rel="icon" type="image/x-icon" href="{{ favicon_url }}"/>
```

**Review by disrupted — COMMENTED** [review]
amazing to see this being picked up so quickly 🎉

**Comment by omarmoo5** [comment]
> amazing to see this being picked up so quickly 🎉

I had this thoughts in one of my projects recently. your issue inspired me to open this PR.

File: starlette_admin/templates/base.html (hunk 1 above)
Comment (omarmoo5) [reply starlette_admin/templates/base.html]: Thanks !

**Review by disrupted — COMMENTED** [review]
I believe the `Admin` class also needs to be updated to include the new keyword argument

**Review by disrupted — APPROVED** [review]
I just tested this and it works great, thanks!

**Comment by jowilf** [comment]
Thank @omarmoo5  for this PR and thanks @disrupted for your reviews
@omarmoo5 You forgot the mongoengine Admin class

File: starlette_admin/contrib/odmantic/admin.py [hunk 2]
```diff
@@ -19,6 +19,7 @@ def __init__(
         route_name: str = "admin",
         logo_url: Optional[str] = None,
         login_logo_url: Optional[str] = None,
+        favicon_url: Optional[str] = None,
```
Comment (jowilf) [inline starlette_admin/contrib/odmantic/admin.py:22]: It's better to move it down to ensure backward compatibility

**Review by jowilf — CHANGES_REQUESTED** [review]
-

File: starlette_admin/contrib/odmantic/admin.py (hunk 2 above)
Comment (omarmoo5) [reply starlette_admin/contrib/odmantic/admin.py:22]: totally agree, I 'll update it.

**Comment by omarmoo5** [comment]
> @omarmoo5 You forgot the mongoengine Admin class

@jowilf  `mongoengine` Admin class doesn't implement it's own constructor, i believe no modification is required there. right ?

**Comment by jowilf** [comment]
> @jowilf mongoengine Admin class doesn't implement it's own constructor, i believe no modification is required there. right ?

You are absolutely right!  I didn't double-check and relied on my memory 😅

**Review by jowilf — APPROVED** [review]
This looks good to me. Can you update the **Unreleased** section in doc/changelog, so I can merge it directly?

**Comment by omarmoo5** [comment]
> This looks good to me. Can you update the **Unreleased** section in doc/changelog, so I can merge it directly?

Thanks, Done.

**Review by jowilf — APPROVED** [review]

**Comment by jowilf** [comment]
Thank you @omarmoo5 for your contribution

**Pull request merged**